src/weightedgauss.jl

```diff
@@ -56,6 +56,9 @@ you may need to decrease `rtol`.   You can also pass in a specialized quadrature
 via the `quad` keyword argument, which should accept arguments `quad(f,a,b,rtol=_,atol=_)`
 similar to `quadgk`.  (This is useful if your weight function has discontinuities, in which
 case you might want to break up the integration interval at the discontinuities.)
+
+The type used for calculations and the return value is the same as the type the
+endpoints `a` and `b`.
```

Comment:
Note that the return type is a floating-point promotion of the type of the endpoints, which may not match the endpoint type.

Also, I don't think we want to guarantee anything about the type used for intermediate calculations (which could conceivably be a higher precision).